apis/v1alpha1/adminnetworkpolicy_types.go

```diff
@@ -136,6 +136,8 @@ type AdminNetworkPolicyIngressRule struct {
 	//
 	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=100
+	// +kubebuilder:validation:XValidation:rule="self.all(value, !has(value.nodes))",message="cluster-ingress traffic controls are unsupported"
+	// +kubebuilder:validation:XValidation:rule="self.all(value, !has(value.externalNetworks))",message="cluster-ingress traffic controls are unsupported"
```

Comment:
This indicates to me that you need a separate type for `From` and `To`. IIUC, you want to add these new fields only in the `To` list, but not the `From`?
If you had a `AdminNetworkPolicyPeerTo` and a separate `AdminNetworkPolicyPeerFrom` this would make things a lot cleaner, eg, you would not generate documentation for fields in `From` that are actually unusable

Reply:
right but in future the From and To will be the same, there is planned efforts for adding ingress support, but at first pass we are doing egress support only. So in 6 months from now, this validation will be gone and from will be same as to. Until then, we want a way to say egress peers only.

Reply:
@JoelSpeed : we discussed this in our upstream meeting and there were no objections in having two sets of definitions like you stated as best practice. So we will go with that and drop the idea of using CELs.